admin-core/src/main/java/com/depromeet/domain/franchise/infrastructure/persistence/FranchiseRepositoryImpl.java

```diff
@@ -26,19 +27,27 @@ public List<Franchise> findAll() {
 
     @Override
     public void save(final Franchise franchise) {
-        final FranchiseEntity franchiseEntity = Franchise.toEntity(franchise);
+        final var franchiseEntity = Franchise.toEntity(franchise);
         franchiseEntityRepository.save(franchiseEntity);
     }
 
     @Override
     public Optional<Franchise> findByName(final String name) {
-        final Optional<FranchiseEntity> findFranchiseEntity = franchiseEntityRepository.findByName(name);
+        final var findFranchiseEntity = franchiseEntityRepository.findByName(name);
         return findFranchiseEntity.map(FranchiseEntity::toDomain);
     }
 
     @Override
-    public Optional<Franchise> findById(long id) {
-        final Optional<FranchiseEntity> findFranchiseEntity = franchiseEntityRepository.findById(id);
-        return findFranchiseEntity.map(FranchiseEntity::toDomain);
+    public Franchise findById(final long id) {
+        final var findFranchiseEntity = franchiseEntityRepository.findById(id)
+                 .orElseThrow(FranchiseNotFoundException::new);
+        return FranchiseEntity.toDomain(findFranchiseEntity);
```

Comment:
정적 메서드를 사용하지 않고, 객체에게 메시지를 보내는 방향으로 할 수 있을 것 같아요
toDomain으로 하면 Franchise 가 가지고 있는 List<Drink> 에 대해서는 초기화를 하지 못 하는데, 이런 경우 toDomainExcludeDrinks() 처럼 Domain 이 가지는 상태에 대해 명확하게 전달하면 어떨까요??

Aggregate Root Domain이 존재한다면, 해당 Domain이 가지는 모든 상태 값을 DB로 부터 load 하는건 성능 문제가 발생할 수 있겠다 판단했고, 필요에 따라 List<Drink> 는 load를 하지 않는 방향도 함께 고민해보면 좋을 것 같아요
```suggestion
        return findFranchiseEntity.toDomain();
```